### PR TITLE
Instagram Challenge

### PR DESCRIPTION
**Pull request by MattGough**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# See https://help.github.com/articles/ignoring-files for more about ignoring files.
+#
+# If you find yourself ignoring temporary files generated by your text editor
+# or operating system, you probably want to add a global ignore instead:
```
Comment (tcpickard94) [inline .gitignore]: Updated README with installation and usage instructions

File: spec/features/posts_feature_spec.rb [hunk 2]
```diff
@@ -0,0 +1,104 @@
+require 'rails_helper'
```
Comment (tcpickard94) [inline spec/features/posts_feature_spec.rb]: requires can be added to your spec_helper

File: spec/features/users_feature_spec.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      click_link('Sign up')
+      fill_in('Email', with: 'test@example.com')
+      fill_in('Password', with: 'testtest')
+      fill_in('Password confirmation', with: 'testtest')
+      click_button('Sign up')
+    end
+
+    it "should see 'sign out' link" do
+      visit('/')
+      expect(page).to have_link('Sign out')
```
Comment (tcpickard94) [inline spec/features/users_feature_spec.rb]: Could use a within block to only test certain parts of the page

File: spec/features/posts_feature_spec.rb [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+  context 'viewing posts' do
+
+    let!(:post1){Post.create(message: 'Hello world!')}
+
+    scenario 'lets a user view a post' do
+      visit '/posts'
+      click_link 'Hello world!'
+      expect(page).to have_content 'Hello world!'
+      expect(page).to have_content "Posted at #{DateTime.now.strftime('%I:%M%p, %m/%d/%Y')}"
```
Comment (tcpickard94) [inline spec/features/posts_feature_spec.rb]: Try and separate expects into single tests

File: app/controllers/comments_controller.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
+class CommentsController < ApplicationController
+  def new
+   @post = Post.find(params[:post_id])
+   @comment = Comment.new
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @post.comments.create(comment_params)
```
Comment (tcpickard94) [inline app/controllers/comments_controller.rb]: Could try a nested form https://github.com/makersacademy/instagram-challenge/blob/master/docs/review.md#use-nested-forms-where-appropriate

File: app/controllers/posts_controller.rb [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    @posts = Post.all.reverse
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.user = current_user
```
Comment (tcpickard94) [inline app/controllers/posts_controller.rb]: This could be done like comments are created

File: app/controllers/posts_controller.rb [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  def edit
+    @post = Post.find(params[:id])
+    if @post.user_id != current_user.id
+      flash[:notice] = 'Cannot edit post belonging to another user'
+      redirect_to '/posts'
+    end
+  end
+
+  def update
+    @post = Post.find(params[:id])
```
Comment (tcpickard94) [inline app/controllers/posts_controller.rb]: Don't need instance variables when redirecting

File: app/controllers/posts_controller.rb [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  def create
+    @post = Post.new(post_params)
+    @post.user = current_user
+    @post.save
+    @posts = Post.all
+    @posts << @post
+    redirect_to '/posts'
+  end
+
+  def post_params
```
Comment (tcpickard94) [inline app/controllers/posts_controller.rb]: This can be a private method

File: app/views/layouts/application.html.erb [hunk 9]
```diff
@@ -0,0 +1,22 @@
+  <!DOCTYPE html>
+  <html>
+  <head>
+    <title>InstagramChallenge</title>
+    <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
+    <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
+    <%= csrf_meta_tags %>
+  </head>
+  <body>
+  <% if user_signed_in? %>
```
Comment (tcpickard94) [inline app/views/layouts/application.html.erb]: Could refactor into a partial

File: app/views/posts/index.html.erb [hunk 10]
```diff
@@ -0,0 +1,22 @@
+<a href="posts/new">Add a post</a>
```
Comment (tcpickard94) [inline app/views/posts/index.html.erb]: Could use semantic html containers i.e. section, main, article

**Comment by mechacoach** [comment]
You had your instagram-challenge reviewed by **Tom Pickard**

### The good points are:

* Consistent indentation, Dead code and comments removed, Unnecessary files removed
* Text expectations appropriate, Test descriptions consistent, Tests DRY
* Models appropriately tested
* Feature tests declarative
* Empty helper files deleted
* Thin controllers with minimal logic
* Model logic is DRY
* Models appropriately connected
* Relative and absolute paths used appropriately
* Asset Path used correctly
* Long nested call chains avoided

### You should consider the following improvements:

* Installation/Usage Instructions should be in README
* Ensure Rakefile has Appropriate Tasks
* Gemfile should use Test Groups
* Ensure spec_helper.rb is Configured Correctly
* Ensure Asset Routes are Set Correctly
* Use nested forms where appropriate
* Avoid generating unnecessary routes using resources in routes.rb
* No need to set instance vars when redirecting
* Feature tests should use 'within' for precision
* Tests should separate out expectations where appropriate
* Use partials appropriately
* Replace instance variables with local variables
* Prefer other Semantic HTML Elements to Divs Where Possible
* Avoid anonymous javascript functions

### Additional comments:
Great job! Think about MVP and what features are necessary to the application

see https://github.com/makersacademy/instagram-challenge/blob/master/docs/review.md for more details

**Pull request closed without merging**